client/cmd/list.go

```diff
@@ -0,0 +1,291 @@
+	if config.IsApiKey() {
+		req.Header.Set("Api-Key", config.Token)
+	}
+	req.Header.Set("User-Agent", fmt.Sprintf("hoopcli/%v", version.Get().Version))
+
+	resp, err := httpclient.NewHttpClient(config.TlsCA()).Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed performing request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	log.Debugf("http response %v", resp.StatusCode)
+
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("failed to fetch connections, status=%v", resp.StatusCode)
+	}
+
+	var connections []map[string]any
+	if err := json.NewDecoder(resp.Body).Decode(&connections); err != nil {
+		return nil, fmt.Errorf("failed decoding response: %v", err)
+	}
+
+	return connections, nil
+}
+
+func displayConnections(connections []map[string]any) {
+	if len(connections) == 0 {
+		fmt.Println("No connections available")
+		return
+	}
+
+	// Fetch agent information
+	config := clientconfig.GetClientConfigOrDie()
+	agentInfo := fetchAgentInfo(config)
+
+	w := tabwriter.NewWriter(os.Stdout, 6, 4, 3, ' ', tabwriter.TabIndent)
+	defer w.Flush()
+
+	if verboseFlag {
+		fmt.Fprintln(w, "NAME\tCOMMAND\tTYPE\tAGENT\tSTATUS\tTAGS\t")
+	} else {
+		fmt.Fprintln(w, "NAME\tTYPE\tAGENT\tSTATUS\t")
+	}
+
+	// Sort connections by name
+	sortConnections(connections)
+
+	for _, conn := range connections {
+		name := toStr(conn["name"])
+		connType := toStr(conn["type"])
+		status := toStr(conn["status"])
+		agentID := toStr(conn["agent_id"])
+
+		// Get agent name
+		agentName := getAgentName(agentInfo, agentID)
+
+		if verboseFlag {
+			cmdList, _ := conn["command"].([]any)
+			command := joinCommand(cmdList)
+			tags := formatConnectionTags(conn["connection_tags"])
+			fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t",
+				name, command, connType, agentName, status, tags)
+		} else {
+			fmt.Fprintf(w, "%s\t%s\t%s\t%s\t",
+				name, connType, agentName, status)
+		}
+		fmt.Fprintln(w)
+	}
+}
+
+func fetchAgentInfo(config *clientconfig.Config) map[string]map[string]any {
+	url := config.ApiURL + "/api/agents"
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		log.Debugf("failed creating agents request: %v", err)
+		return nil
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %v", config.Token))
+	if config.IsApiKey() {
+		req.Header.Set("Api-Key", config.Token)
+	}
+	req.Header.Set("User-Agent", fmt.Sprintf("hoopcli/%v", version.Get().Version))
+
+	resp, err := httpclient.NewHttpClient(config.TlsCA()).Do(req)
+	if err != nil {
+		log.Debugf("failed performing agents request: %v", err)
+		return nil
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		log.Debugf("failed to fetch agents, status=%v", resp.StatusCode)
+		return nil
+	}
+
+	var agents []map[string]any
+	if err := json.NewDecoder(resp.Body).Decode(&agents); err != nil {
+		log.Debugf("failed decoding agents response: %v", err)
+		return nil
+	}
```

Comment:
Make sure to return a generic error for calls to the api, this will make sure that we all catch any non success error along with the body, example:

```sh
	resp, err := httpclient.NewHttpClient(config.TlsCA()).Do(req)
	if err != nil {
		return nil, nil, err
	}
	log.Debugf("http response %v", resp.StatusCode)
	defer resp.Body.Close()
	if resp.StatusCode != 200 {
		respBody, _ := io.ReadAll(resp.Body)
		return nil, resp.Header, fmt.Errorf("failed performing request, status=%v, body=%v",
			resp.StatusCode, string(respBody))
	}
```